### PR TITLE
feat(Pivot): add overflowButtonAs prop

### PR DESCRIPTION
**Pull request by alina-zaieva**
## Current Behavior

Not applicable, this is a new functionality.

## New Behavior

* `oveflowButtonAs` prop is added to `Pivot` component
* When `overflow` is set to `menu` and `overflowButtonAs` is provided, the component will be used for rendering overflow button instead of default `CommandButton`

Usage example: setting custom label and icon for the overflow button
image

## Related Issue(s)

Implements #24001

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/20fd0422e9831a560717f39a8d0cbdf57bc721ca'>20fd0422e9831a560717f39a8d0cbdf57bc721ca</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/BaseButton-mount_base.html">961</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/BaseButton-mount.html">982</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Breadcrumb-mount_base.html">2848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Breadcrumb-mount.html">2841</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Checkbox-mount_base.html">2733</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Checkbox-mount.html">2615</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/CheckboxBase-mount_base.html">2334</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/CheckboxBase-mount.html">2380</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ChoiceGroup-mount_base.html">4836</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ChoiceGroup-mount.html">4856</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ComboBox-mount_base.html">982</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ComboBox-mount.html">1037</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/CommandBar-mount_base.html">11047</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/CommandBar-mount.html">11209</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ContextualMenu-mount_base.html">12922</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ContextualMenu-mount.html">12959</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DefaultButton-mount_base.html">1172</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DefaultButton-mount.html">1209</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DetailsRow-mount_base.html">3968</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DetailsRow-mount.html">4056</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DetailsRowFast-mount_base.html">3993</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DetailsRowFast-mount.html">3995</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DetailsRowNoStyles-mount_base.html">3765</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DetailsRowNoStyles-mount.html">3860</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Dialog-mount_base.html">2928</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Dialog-mount.html">2979</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DocumentCardTitle-mount_base.html">185</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/DocumentCardTitle-mount.html">175</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Dropdown-mount_base.html">3448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Dropdown-mount.html">3538</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/FocusTrapZone-mount_base.html">1962</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/FocusTrapZone-mount.html">1925</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/FocusZone-mount_base.html">1847</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/FocusZone-mount.html">1918</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/IconButton-mount_base.html">1823</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/IconButton-mount.html">1765</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Label-mount_base.html">357</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Label-mount.html">390</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Layer-mount_base.html">3238</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Layer-mount.html">3315</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Link-mount_base.html">492</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Link-mount.html">504</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/MenuButton-mount_base.html">1572</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/MenuButton-mount.html">1609</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/MessageBar-mount_base.html">2259</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/MessageBar-mount.html">2291</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Nav-mount_base.html">3407</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Nav-mount.html">3398</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/OverflowSet-mount_base.html">1119</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/OverflowSet-mount.html">1134</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Panel-mount_base.html">2322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Panel-mount.html">2299</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Persona-mount_base.html">1034</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Persona-mount.html">1039</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Pivot-mount_base.html">1462</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Pivot-mount.html">1511</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/PrimaryButton-mount_base.html">1335</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/PrimaryButton-mount.html">1401</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Rating-mount_base.html">8007</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Rating-mount.html">7987</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SearchBox-mount_base.html">1343</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SearchBox-mount.html">1352</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Shimmer-mount_base.html">2584</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Shimmer-mount.html">2579</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Slider-mount_base.html">2062</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Slider-mount.html">2056</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SpinButton-mount_base.html">5160</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SpinButton-mount.html">5126</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Spinner-mount_base.html">445</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Spinner-mount.html">441</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SplitButton-mount_base.html">3409</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SplitButton-mount.html">3360</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Stack-mount_base.html">536</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Stack-mount.html">547</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/StackWithIntrinsicChildren-mount_base.html">2386</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/StackWithIntrinsicChildren-mount.html">2336</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/StackWithTextChildren-mount_base.html">5389</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/StackWithTextChildren-mount.html">5339</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SwatchColorPicker-mount_base.html">11921</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/SwatchColorPicker-mount.html">11871</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/TagPicker-mount_base.html">2833</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/TagPicker-mount.html">2803</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/TeachingBubble-mount_base.html">102207</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/TeachingBubble-mount.html">101797</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Text-mount_base.html">443</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Text-mount.html">430</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/TextField-mount_base.html">1453</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/TextField-mount.html">1491</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ThemeProvider-mount_base.html">1286</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ThemeProvider-mount.html">1310</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ThemeProvider-virtual-rerender_base.html">683</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ThemeProvider-virtual-rerender.html">731</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1932</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1919</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Toggle-mount_base.html">849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/Toggle-mount.html">842</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/buttonNative-mount_base.html">146</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24196/perf-test/buttonNative-mount.html">127</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: change/@fluentui-react-8eaa0012-dc28-494f-a9c8-41bbb8f24c1a.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add overflowButtonAs prop to Pivot component",
```
Comment (TristanWatanabe) [inline change/@fluentui-react-8eaa0012-dc28-494f-a9c8-41bbb8f24c1a.json:3]:
```suggestion
  "comment": "feat(Pivot): add overflowButtonAs prop to allow overflow button customization.",
```

**Review by TristanWatanabe — APPROVED** [review]
Minor nit on the changefile but otherwise this LGTM!

File: change/@fluentui-react-8eaa0012-dc28-494f-a9c8-41bbb8f24c1a.json (hunk 1 above)
Comment (alina-zaieva) [reply change/@fluentui-react-8eaa0012-dc28-494f-a9c8-41bbb8f24c1a.json]: Applied the change 🙂

**Comment by alina-zaieva** [comment]
@TristanWatanabe @behowell can this PR be merged now?

**Comment by TristanWatanabe** [comment]
@alina-zaieva Going to merge this in now - thanks for contributing to the library!

**Pull request merged**